nessie/src/main/java/org/apache/iceberg/nessie/NessieIcebergClient.java

```diff
@@ -523,4 +528,49 @@ public void close() {
       api.close();
     }
   }
+
+  public TableMetadata loadTableMetadataWithNessieSpecificProperties(
+      String metadataLocation, IcebergTable table, FileIO fileIO, String identifier) {
+    // Update the TableMetadata with the Content of NessieTableState.
+    TableMetadata deserialized = TableMetadataParser.read(fileIO, metadataLocation);
```

Comment:
Caller handling the table metadata file reading will leads to duplicate code in multiple places. 

Since it is a static method in Util now, I think ok to pass the FileIO.

Reply:
I agree with @dimas-b here. I don't think calling `TableMetadataParser.read(fileIO, metadataLocation)` should actually be in a utility method. There should rather be a method that takes `TableMetadata` and just adds whatever properties Nessie has to set

Reply:
@dimas-b, @nastra:

https://github.com/apache/iceberg/blob/9f12b85667362dbf0cc1bbb16972602a103938a8/core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java#L208C38-L208C38

Currently, `metadataLoader` just reads the tableMetadata for other catalogs and it is under the retry as shown above. 
If I read the file before and change `metadataLoader` to dummy for other catalogs, it will be just an assignment operation for other catalogs and may not need the retry. 

I am not sure about the impact of removing the retry here. WDYT?

Reply:
Got it. I guess I was overthinking with changing the `metadataLoader` of `refreshFromMetadataLocation`. 

Handled and pushed as a separate commit.